### PR TITLE
Choose method

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented choose method.

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -608,7 +608,34 @@ cdef class ndarray:
         """
         return _repeat(self, repeats, axis)
 
-    # TODO(okuta): Implement choose
+    cpdef choose(self, choices, out=None, mode='raise'):
```
Comment (gwtnb) [inline cupy/core/core.pyx]: Is this `out` unused?

Comment (unnonouno) [inline cupy/core/core.pyx]: oh

File: cupy/core/core.pyx [hunk 2]
```diff
@@ -608,7 +608,37 @@ cdef class ndarray:
...
+        ba, bcs = broadcast(a, choices).values
+
+        if out is None:
+            out = ndarray(ba.shape[1:], choices.dtype)
+
+        n_channel = numpy.prod(bcs[0].shape)
+        if mode == 'raise':
+            if not ((a < n).all() and (0 <= a).all()):
+                raise ValueError('invalid entry in choice array')
+            c = _choose_kernel(ba[0], bcs, n_channel, out)
```
Comment (gwtnb) [inline cupy/core/core.pyx]: `c` seems unnecessary.

**Comment by gwtnb** [comment]
LGTM!

**Pull request merged**